### PR TITLE
AppTPCompanyDetailsAdapter, index out of bounds exceptions

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/0/1205530419817941/f

### Description
Fixed crash in AppTPCompanyDetailsAdapter.

### Steps to test this PR

- [ ] Install from this branch.
- [ ] Enable AppTP and wait to have some trackers blocked.
- [ ] Tap on one of the trackers to see the company details. 
- [ ] Toggle OFF/ON to disable/enable protections from the button positioned on the top right. The app should not crash.

### CODE REVIEW

**Comment by anikiki** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3558** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3558" target="_blank">Graphite</a> 
    * **PR #3560** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3560" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3560?utm_source=stack-comment).

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/AppTPCompanyTrackersViewModel.kt [hunk 1]
```diff
@@ -150,25 +150,23 @@ class AppTPCompanyTrackersViewModel @Inject constructor(
...
-                val protectionState = excludedAppsRepository.getAppProtectionStatus(packageName)
-
-                viewStateFlow.emit(
-                    viewStateFlow.value.copy(
-                        toggleChecked = protectionState == PROTECTED,
-                        bannerState = protectionState.getBannerState(),
-                        toggleEnabled = protectionState != UNPROTECTED_THROUGH_NETP,
-                    ),
-                )
+            if (checked) {
```
Comment (anikiki) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/AppTPCompanyTrackersViewModel.kt:153]: Removed `withContext(dispatchers.io())` as it's not needed.

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/AppTPCompanyTrackersActivity.kt [hunk 2]
```diff
@@ -185,7 +185,7 @@ class AppTPCompanyTrackersActivity : DuckDuckGoActivity() {
         )
         binding.includeToolbar.appTrackedAgo.text = viewState.lastTrackerBlockedAgo
 
-        lifecycleScope.launch(dispatcherProvider.io()) {
+        lifecycleScope.launch {
```
Comment (anikiki) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/AppTPCompanyTrackersActivity.kt:188]: Reverted this change.

**Comment by marcosholgado** [comment]
@anikiki is this ready to review?

**Comment by anikiki** [comment]
Closing this as there is another approach in https://github.com/duckduckgo/Android/pull/3747

**Pull request closed without merging**